### ISSUE TITLE
Additional Log in the softmax function

### ISSUE THREAD

**Issue by ATAboukhadra** [AberHu/Knowledge-Distillation-Zoo#12]
Hi, 

I just have a question, why did you use an additional log before the softmax function in st.py loss on one input but not the other?

```
def forward(self, out_s, out_t):
		loss = F.kl_div(F.log_softmax(out_s/self.T, dim=1),
						F.softmax(out_t/self.T, dim=1),
						reduction='batchmean') * self.T * self.T

		return loss

```

**Comment by nguyenvulong** [comment]
Yeah it is strange that is has the `log` there. I found [another implementation](https://github.com/SforAiDl/KD_Lib/blob/9b29607913/KD_Lib/KD/vision/vanilla/vanilla_kd.py) that did not use `log` but `softmax` only for both Teacher and Student.


As you might have already known it, when we modify the  `F.log_softmax(out_s/self.T, dim=1)` in to `F.softmax(out_s/self.T, dim=1)`, the `KD` output is **negative**. 

Do you have any ideas why?

<img width="1877" alt="image" src="https://user-images.githubusercontent.com/1311412/177104615-6df9104c-b54b-4047-8cae-134b95396d1a.png">

**Comment by nguyenvulong** [comment]
I asked another researcher and here is the answer https://github.com/yoshitomo-matsubara/torchdistill/issues/233
